src/Uno.UWP/Storage/KnownFolders.Android.cs

```diff
@@ -1,14 +1,101 @@
 ﻿#if __ANDROID__
 
+using System;
+using System.IO;
+
 namespace Windows.Storage
 {
 	public static partial class KnownFolders
 	{
-		internal static StorageFolder FolderFromAndroidName(string name) => new StorageFolder(Android.OS.Environment.GetExternalStoragePublicDirectory(name).CanonicalPath);
-		public static StorageFolder MusicLibrary => FolderFromAndroidName(Android.OS.Environment.DirectoryMusic); 
-		public static StorageFolder VideosLibrary => FolderFromAndroidName(Android.OS.Environment.DirectoryMovies); 
+		internal static void CheckPermission()

```

Comment:
This method is generic and resembles the code present here:

https://github.com/unoplatform/uno/blob/6e8a9273acbd6fcab385fc36a3d29cb48295f824/src/Uno.UI/Extensions/PermissionsHelper.Android.cs

Can you validate that you're not doing duplicate work ? Maybe the permission helper should be moved.